filament/backend/src/opengl/PlatformWGL.cpp

```diff
@@ -183,8 +183,22 @@ Platform::SwapChain* PlatformWGL::createSwapChain(void* nativeWindow, uint64_t&
 }
 
 Platform::SwapChain* PlatformWGL::createSwapChain(uint32_t width, uint32_t height, uint64_t& flags) noexcept {
-    // TODO: implement headless SwapChain
-    return nullptr;
+    // WS_POPUP was chosen for the window style here after some experimentation.
+    // For some reason, using other window styles resulted in corrupted pixel buffers when using
+    // readPixels.
+    RECT rect = {0, 0, width, height};
+    AdjustWindowRect(&rect, WS_POPUP, FALSE);
+    width = rect.right - rect.left;
+    height = rect.bottom - rect.top;
+
+    // TODO: this window never gets destroyed.
+    HWND hWnd = CreateWindowA("STATIC", "headless", WS_POPUP, 0, 0,

```

Comment:
Could we make this a member of the SwapChain so we can destroy it when the SwapChain is destroy for instance? This window could be large, we really need to clean it up.

Reply:
I agree it's important because typically we will be running unit tests headless, could be many many in a row.